File: backend/src/main/java/com/ums/ums_backend/entity/Department.java
package com.ums.ums_backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@Setter
public class Department {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "faculty_id", nullable = false)
    private Faculty faculty;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(nullable = false, length = 10)
    private String code;

    @Column(length = 250)
    private String description;

    @OneToMany(mappedBy = "department")
    private List<Course> courses;

}

File: frontend/src/app/features/department/models/department.ts
import {Faculty} from '../../faculty/models/faculty';
import {Course} from '../../course/models/course';

export interface Department {

  id: number,
  faculty: Faculty,
  name: string,
  code: string,
  description: string,
  courses: Course[]

}

export interface DepartmentCreateRequest {

  facultyId: number,
  name: string,
  code: string,
  description: string,

}

export interface DepartmentCreateRequest {

  facultyId: number,
  name: string,
  code: string,
  description: string,

}
